lib/ocpp/v201/database_handler.cpp

```diff
@@ -634,5 +634,19 @@ void DatabaseHandler::insert_or_update_charging_profile(const int evse_id, const
     }
 }
 
+void DatabaseHandler::delete_charging_profile(const int profile_id) {
+    std::string sql = "DELETE FROM CHARGING_PROFILES WHERE ID = @profile_id;";
```

Comment:
Just a comment: It occurs to me that I'm used to seeing more assistance from the DB interface to craft an SQL statement that is safe, and I know there are ways to have injection attacks and such if you don't take care to sanitize input. I imagine there isn't much here (it's just an int), but I'm surprised there isn't something to leverage to avoid writing entire statements by hand.

Reply:
You are correct when crafting SQL statements there is always a concern for SQL Injection. I assume when you say interface you are referring to an ORM of sorts? Since we are on minimal hardware that would be a bit resource intensive. The pattern here is using prepared statements and binding the parameters see line 641. In this way the values are only treated as data and ensure the security.